reascripts/ReaSpeech/source/TranscriptExporter.lua

```diff
@@ -34,23 +34,119 @@ function TranscriptExporter:init()
     TranscriptExportFormat.exporter_srt(),
     TranscriptExportFormat.exporter_csv(),
   }
+  function self.export_formats.on_change()
+    self:update_target_filename_ui()
+  end
+
   self.export_options = {}
+
   self.file_selector = ReaSpeechFileSelector.new({
     label = 'File',
     save = true,
     button_width = self.BUTTON_WIDTH,
-    input_width = self.FILE_WIDTH
+    input_width = self.FILE_WIDTH,
+    on_set = function() self:update_target_filename_ui() end,
   })
 
+  -- invisible state to manage the UX around the target filename
+  self.target_filename_exists = Storage.memory(false)
+  self.has_extension = Storage.memory(false)
+
+  -- this is the value that will actually be used after the
+  -- file_selector contents are processed (relative vs full path,
+  -- auto extension, etc.)
+  self.target_filename = Storage.memory('')
+  self.target_filename_display = Storage.memory('')
+
+  self.apply_extension = ReaSpeechCheckbox.new {
+    default = true,
+    label_long = 'Apply Extension',
+    disabled_if = function() return self.has_extension:get() end,
+    changed_handler = function()
+      if not self.apply_extension then return end
+      self:update_target_filename_ui()
+    end
+  }
+
   self.alert_popup = AlertPopup.new {}
 end
 
+function TranscriptExporter:open()
+  self:reset_form()
+end
+
+function TranscriptExporter:reset_form()
+  self.file_selector:set('')
+  self.apply_extension:set(true)
+  self:update_target_filename_ui()
+end
+
+function TranscriptExporter:clear_target_filename()
+  self.target_filename:set('')
+  self.target_filename_display:set('')
+  self.target_filename_exists:set(false)
+  self.has_extension:set(false)
+end
+
+function TranscriptExporter:update_target_filename_ui()
+  local specified_path = self.file_selector:value()
+
+  if specified_path == '' then
+    self:clear_target_filename()
+    return
+  end
+
+  local is_full_path = PathUtil.is_full_path(specified_path)
+
+  local full_path = PathUtil.get_real_path(specified_path)
+
+  self.has_extension:set(PathUtil.has_extension(full_path))
+
+  -- automatically turn off the auto-extension if user enters
+  -- in a filename with an extension
+  if self.has_extension:get() and self.apply_extension:value() then
+    self.apply_extension:set(false)
+  end
+
+  -- automatically apply default extension for format, if desired
+  if self.apply_extension:value() then
+    local extension = self.export_formats:selected_format().extension
+    full_path = PathUtil.apply_extension(full_path, extension)
+  end
+
+  self.target_filename:set(full_path)
+
+  if is_full_path then
+    self.target_filename_display:set(full_path)
+  else
+    local display = table.concat({
+      '<Project Resources>',
+      full_path:sub(#reaper.GetProjectPath() + 2)
+    }, PathUtil._path_separator())
```

Comment:
Python has `os.path.join`, which I've found helpful for stuff like this.

Reply:
noticing myself calling a nominally-private method on `PathUtil` was a hint in that direction lol